numba_cuda/numba/cuda/dispatcher.py

```diff
@@ -848,7 +848,12 @@ def typeof_pyval(self, val):
         # Based on _DispatcherBase.typeof_pyval, but differs from it to support
         # the CUDA Array Interface.
         try:
-            return typeof(val, Purpose.argument)
+            tp = typeof(val, Purpose.argument)
```

Comment:
I'm concerned this could have a non-trivial impact on kernel launch time. Can you do a microbenchmark to check how much this impacts the latency of launches with various numbers of array arguments?